echo2_integration_test.cc

```diff
@@ -1,7 +1,42 @@
+#include <type_traits>
+
 #include "test/integration/integration.h"
 #include "test/integration/utility.h"
 
 namespace Envoy {
+namespace {
+
+// NoCreateConnectionDriver is used to provide "value" member regardless `createConnectionDriver`
```

Comment:
This is too complicated. Please just remove all of this and put in the final code that you need. We can force merge it here, and then you can update the SHA of filter example in your other PR. Thank you!

Reply:
Sounds good. It's a great c++ template practice in the weekend though.

I am reverting this PR